### PR TITLE
✨ Update home page

### PR DESCRIPTION
**Pull request by YosefAshenafi**
# Status
WIP

# Features

- Update header Menus list
- Add home page carousel as starter

# Screenshot
contrarian

### CODE REVIEW

File: src/app/components/Header/index.tsx [hunk 1]
```diff
@@ -1,21 +1,29 @@
...
-              <Nav className="justify-content-end flex-grow-1 pe-3">
-                <Nav.Link href="/">Home</Nav.Link>
+              <Nav className="justify-content-start flex-grow-1 pe-3">
+                {MENU.map(function (menu) {
+                  return (
+                    <Nav.Link className="mx-5" href="/">
+                      {menu.name}
+                    </Nav.Link>
+                  )
+                })}
```
Comment (gwainor) [inline src/app/components/Header/index.tsx]: Please pay attention to console errors. There should be no need to tell you these basic things anymore. If you are mapping a list you **must** add `key` prop!
```suggestion
                {MENU.map ((menu) => (
                    <Nav.Link className="mx-5" href={menu.link} key={menu.name}>
                      {menu.name}
                    </Nav.Link>
                ))}
```

File: public/home-background.png
Comment (gwainor) [inline public/home-background.png:1]: Please put images inside `public/images` directory

File: src/app/components/Home/carousel.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import { CarouselLists } from '@/app/constants'
+import Image from 'next/image'
+import Carousel from 'react-bootstrap/Carousel'
+
+export function HomepageCarousel() {
+  return (
+    <div>
+      <Carousel fade>
+        {CarouselLists.map(function (item) {
+          return (
```
Comment (gwainor) [inline src/app/components/Home/carousel.tsx]: Use arrow function please. Do not forget `key` prop.

File: src/app/layout.tsx [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
-import '@/app/scss/style.scss'
 import { Header } from '@/app/components/Header'
+import '@/app/scss/style.scss'
+import type { Metadata } from 'next'
+import { Merriweather, Montserrat } from 'next/font/google'
 
 const montserrat = Montserrat({
   subsets: ['latin'],
-  weight: ['400', '700'],
+  // weight: ['400', '700'],
```
Comment (gwainor) [inline src/app/layout.tsx]: What is the reason for commenting this?

File: src/app/page.tsx [hunk 4]
```diff
@@ -1,17 +1,12 @@
 'use client'
 import { Container } from 'react-bootstrap'
+import { HomepageCarousel } from './components/Home/carousel'
```
Comment (gwainor) [inline src/app/page.tsx]:
```suggestion
import { HomepageCarousel } from '@/components/Home/carousel'
```

File: src/app/scss/style.scss [hunk 5]
```diff
@@ -50,7 +50,7 @@
 // @import "../../../node_modules/bootstrap/scss/modal";
 // @import "../../../node_modules/bootstrap/scss/tooltip";
 // @import "../../../node_modules/bootstrap/scss/popover";
-// @import "../../../node_modules/bootstrap/scss/carousel";
+@import "../../../node_modules/bootstrap/scss/carousel";
```
Comment (gwainor) [inline src/app/scss/style.scss]: Move this to after line 36 please

**Review by gwainor — CHANGES_REQUESTED** [review]

File: src/app/components/Header/index.tsx [hunk 6]
```diff
@@ -1,21 +1,29 @@
...
-import { SITE_NAME } from '@/app/constants'
 
 export function Header() {
   return (
     <header>
       <Navbar expand="lg" className="bg-body-tertiary">
         <Container>
-          <Navbar.Brand href="/">{SITE_NAME}</Navbar.Brand>
+          <Navbar.Brand href="/">
+            <h1>{SITE_NAME}</h1>
```
Comment (gwainor) [inline src/app/components/Header/index.tsx]: `h1` is incorrect here. Brand is not heading. You can change font size of navbar brand using SCSS variables.
https://getbootstrap.com/docs/5.3/components/navbar/#sass-variables

If you are not confident enough with a framework, you should read their documentation more.

File: src/app/components/Home/carousel.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
+'use client'
+import { CarouselLists } from '@/app/constants'
+import Image from 'next/image'
+import Carousel from 'react-bootstrap/Carousel'
```
Comment (gwainor) [inline src/app/components/Home/carousel.tsx]:
```suggestion
import { Carousel } from 'react-bootstrap'
```

File: src/app/constants/general.ts [hunk 8]
```diff
@@ -1 +1,43 @@
...
+  {
+    name: 'Contact Us',
+    link: '/',
+  },
+]
+
+export const CarouselLists = [
+  {
+    name: 'Frank Zappa',
+    quote: 'So many books, so little time',
```
Comment (gwainor) [inline src/app/constants/general.ts]:
```suggestion
    quote: 'So many books, so little time.',
```

File: src/app/constants/general.ts [hunk 9]
```diff
@@ -1 +1,43 @@
 export const SITE_NAME = 'Contrarian'
+
+export const MENU = [
```
Comment (gwainor) [inline src/app/constants/general.ts]: `NAVIGATION_LINKS` might be better

File: src/app/constants/general.ts [hunk 10]
```diff
@@ -1 +1,43 @@
...
+  {
+    name: 'Home',
+    link: '/',
+  },
+  {
+    name: 'Books Review',
```
Comment (gwainor) [inline src/app/constants/general.ts]:
```suggestion
    name: 'Book Reviews',
```

File: src/app/constants/general.ts [hunk 11]
```diff
@@ -1 +1,43 @@
...
+    link: '/',
+  },
+  {
+    name: 'Testemonials',
```
Comment (gwainor) [inline src/app/constants/general.ts]:
```suggestion
    name: 'Testimonials',
```

File: src/app/constants/general.ts [hunk 12]
```diff
@@ -1 +1,43 @@
...
+    link: '/',
+  },
+  {
+    name: 'About Us',
+    link: '/',
+  },
+  {
+    name: 'Contact Us',
+    link: '/',
+  },
```
Comment (gwainor) [inline src/app/constants/general.ts]: I don't think these will be needed.

File: src/app/scss/style.scss
Comment (gwainor) [inline src/app/scss/style.scss:1]: Remove this file. `scss` folder moved to `src`

File: src/components/Home/carousel.tsx
Comment (gwainor) [inline src/components/Home/carousel.tsx:1]: Please name this file to `src/components/Carousel/index.tsx`

File: src/app/components/Header/index.tsx
Comment (gwainor) [inline src/app/components/Header/index.tsx:1]: This file moved outside of `app` folder

**Review by gwainor — CHANGES_REQUESTED** [review]

File: src/app/components/Header/index.tsx (hunk 1 above)
Comment (YosefAshenafi) [reply src/app/components/Header/index.tsx]: done

File: public/home-background.png
Comment (YosefAshenafi) [reply public/home-background.png:1]: done

File: src/app/components/Home/carousel.tsx (hunk 2 above)
Comment (YosefAshenafi) [reply src/app/components/Home/carousel.tsx]: done

File: src/app/layout.tsx (hunk 3 above)
Comment (YosefAshenafi) [reply src/app/layout.tsx]: reverted, it used to fix the font on my dev.

File: src/app/page.tsx (hunk 4 above)
Comment (YosefAshenafi) [reply src/app/page.tsx]: done

File: src/app/scss/style.scss (hunk 5 above)
Comment (YosefAshenafi) [reply src/app/scss/style.scss]: done

File: src/app/components/Header/index.tsx (hunk 6 above)
Comment (YosefAshenafi) [reply src/app/components/Header/index.tsx]: Ok, I now fixed it with inline bootstrap font size. Thanks.

File: src/app/components/Home/carousel.tsx (hunk 7 above)
Comment (YosefAshenafi) [reply src/app/components/Home/carousel.tsx]: done

File: src/app/constants/general.ts (hunk 8 above)
Comment (YosefAshenafi) [reply src/app/constants/general.ts]: done

File: src/app/constants/general.ts (hunk 9 above)
Comment (YosefAshenafi) [reply src/app/constants/general.ts]: done

File: src/app/constants/general.ts (hunk 10 above)
Comment (YosefAshenafi) [reply src/app/constants/general.ts]: done

File: src/app/constants/general.ts (hunk 11 above)
Comment (YosefAshenafi) [reply src/app/constants/general.ts]: done

File: src/app/constants/general.ts (hunk 12 above)
Comment (YosefAshenafi) [reply src/app/constants/general.ts]: Ok, Removed

File: src/app/scss/style.scss
Comment (YosefAshenafi) [reply src/app/scss/style.scss:1]: done

File: src/app/components/Header/index.tsx
Comment (YosefAshenafi) [reply src/app/components/Header/index.tsx:1]: done

File: src/components/Home/carousel.tsx
Comment (YosefAshenafi) [reply src/components/Home/carousel.tsx:1]: done

File: src/app/page.tsx [hunk 13]
```diff
@@ -1,17 +1,13 @@
...
-        doloremque dolorem totam hic sapiente rem culpa aut autem quibusdam.
-        Atque!
-      </p>
-      <div style={{ height: '2000px' }}></div>
-    </Container>
+    <main>
+      <Container>
+        <HomepageCarousel />
+      </Container>
+    </main>
```
Comment (gwainor) [inline src/app/page.tsx]: `<main>` is already defined in `layout` we can remove from here:
```suggestion
      <Container>
        <HomepageCarousel />
      </Container>
```

File: src/components/Carousel/index.tsx [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+import { CarouselLists } from '@/constants'
+import Image from 'next/image'
+import { Carousel } from 'react-bootstrap'
+
+export function HomepageCarousel() {
+  return (
+    <div>
+      <Carousel fade>
+        {CarouselLists.map(function (item) {
+          return (
```
Comment (gwainor) [inline src/components/Carousel/index.tsx]: I thought I request you to use arrow function. Also could we rename the constant name to `CAROUSEL_ITEMS`?
```suggestion
        {CarouselLists.map((item) => (
```

File: src/components/Header/index.tsx [hunk 15]
```diff
@@ -39,11 +39,13 @@ export function Header({ theme, toggleTheme }: Props) {
...
         <Container>
-          <Navbar.Brand href="/">{SITE_NAME}</Navbar.Brand>
+          <Navbar.Brand className="fs-1 fw-bold mb-2" href="/">
+            {SITE_NAME}
+          </Navbar.Brand>
           <Navbar.Toggle aria-controls="main-navbar-nav" />
           <Navbar.Offcanvas id="main-navbar-nav" placement="end">
             <Offcanvas.Header closeButton>
-              <Offcanvas.Title className="d-flex align-items-center">
+              <Offcanvas.Title className="d-flex align-items-center ">
```
Comment (gwainor) [inline src/components/Header/index.tsx]:
```suggestion
              <Offcanvas.Title className="d-flex align-items-center">
```

File: src/components/Carousel/index.tsx [hunk 16]
```diff
@@ -0,0 +1,47 @@
+'use client'
+import { CarouselLists } from '@/constants'
+import Image from 'next/image'
+import { Carousel } from 'react-bootstrap'
```
Comment (gwainor) [inline src/components/Carousel/index.tsx]:
```suggestion
import { Carousel as BsCarousel } from 'react-bootstrap'
```

File: src/components/Carousel/index.tsx [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+
+export function HomepageCarousel() {
```
Comment (gwainor) [inline src/components/Carousel/index.tsx]:
```suggestion
export function Carousel() {
```

**Review by gwainor — CHANGES_REQUESTED** [review]

File: src/app/components/Header/index.tsx (hunk 6 above)
Comment (gwainor) [reply src/app/components/Header/index.tsx]: This is not what I requested @YosefAshenafi. Please read what I wrote carefully.

File: src/components/Carousel/index.tsx (hunk 17 above)
Comment (YosefAshenafi) [reply src/components/Carousel/index.tsx]: done

File: src/components/Carousel/index.tsx (hunk 16 above)
Comment (YosefAshenafi) [reply src/components/Carousel/index.tsx]: done

File: src/components/Header/index.tsx (hunk 15 above)
Comment (YosefAshenafi) [reply src/components/Header/index.tsx]: Fixed

File: src/components/Carousel/index.tsx (hunk 14 above)
Comment (YosefAshenafi) [reply src/components/Carousel/index.tsx]: done

File: src/app/page.tsx (hunk 13 above)
Comment (YosefAshenafi) [reply src/app/page.tsx]: Ok, done.

File: src/components/Carousel/index.tsx [hunk 18]
```diff
@@ -0,0 +1,45 @@
+'use client'
+import { CAROUSEL_ITEMS } from '@/constants'
+import Image from 'next/image'
+import { Carousel as BsCarousel } from 'react-bootstrap'
+
+export function Carousel() {
+  return (
+    <div>
```
Comment (gwainor) [inline src/components/Carousel/index.tsx]: Remove this wrapper div please. Sorry I missed this one earlier

File: src/components/Header/index.tsx [hunk 19]
```diff
@@ -1,8 +1,8 @@
 'use client'
+import { NAVIGATION_LINKS, SITE_NAME } from '@/constants'
+import React from 'react'
 import { Container, Nav, Navbar, Offcanvas } from 'react-bootstrap'
```
Comment (gwainor) [inline src/components/Header/index.tsx]:
```suggestion
import React from 'react'
import { NAVIGATION_LINKS, SITE_NAME } from '@/constants'
import { Container, Nav, Navbar, Offcanvas } from 'react-bootstrap'
```

**Review by gwainor — APPROVED** [review]

File: src/components/Header/index.tsx (hunk 19 above)
Comment (YosefAshenafi) [reply src/components/Header/index.tsx]: done, but, it is updated on save.

File: src/components/Carousel/index.tsx (hunk 18 above)
Comment (YosefAshenafi) [reply src/components/Carousel/index.tsx]: done

**Pull request closed without merging**